core/src/test/java/io/grpc/CallOptionsTest.java

```diff
@@ -108,27 +114,39 @@ public void mutateExecutor() {
 
   @Test
   public void testWithDeadlineAfter() {
-    long deadline = CallOptions.DEFAULT
-        .withDeadlineAfter(1, TimeUnit.MINUTES).getDeadlineNanoTime();
-    long expected = System.nanoTime() + 1L * 60 * 1000 * 1000 * 1000;
+    Deadline deadline = CallOptions.DEFAULT.withDeadlineAfter(1, MINUTES).getDeadline();
+    long expected = MINUTES.toNanos(1);
     // 10 milliseconds of leeway
-    long epsilon = 1000 * 1000 * 10;
-
-    assertEquals(expected, deadline, epsilon);
+    long epsilon = MILLISECONDS.toNanos(10);
+    assertNotNull(deadline);
+    assertEquals(expected, deadline.timeRemaining(NANOSECONDS), epsilon);
   }
 
   @Test
   public void testToString() {
-    assertEquals("CallOptions{deadlineNanoTime=null, authority=null}",
-        CallOptions.DEFAULT.toString());
+    assertEquals("CallOptions{deadline=null, authority=null, "
+        + "affinity={}, executor=null, compressorName=null}", CallOptions.DEFAULT.toString());
+
     // Deadline makes it hard to check string for equality.
-    assertEquals("CallOptions{deadlineNanoTime=null, authority=authority}",
-        allSet.withDeadlineNanoTime(null).toString());
-    assertTrue(allSet.toString().contains("deadlineNanoTime=" + sampleDeadlineNanoTime + ","));
+    assertEquals("CallOptions{deadline=null, authority=authority, "
```

Comment:
Ah.  In that case this line should be changed to check for absence rather than exact string matching.